narwhal/primary/src/proposer.rs

```diff
@@ -401,8 +457,11 @@ impl Proposer {
                 }
 
                 // Receive digests from our workers.
-                Some(digest_record) = self.rx_our_digests.recv() => {
+                Some((digest_record, ack_sender)) = self.rx_our_digests.recv() => {
                     self.digests.push(digest_record);
+                    // Signal back to the worker that the batch is recorded on the
```

Comment:
I wrote this:
```
                    // Signal back to the worker that the batch is recorded on the
                    // primary, and will be tracked until inclusion. This means that 
                    // if the primary does not fail it will attempt to send the digest
                    // (and re-send if necessary) until it is sequenced, or the end of
                    // the epoch is reached. For the moment this does not persist primary
                    // crashes and re-starts.
```